src/imio/helpers/date.py

```diff
@@ -0,0 +1,164 @@
+        date = DateTime()
+
+    if long_format:
+        hour = ' (%H:%M)'
+    else:
+        hour = ''
+    #check if we need to return the month name
+    if month_name:
+        month = date.month()
+        translated_month = translate(msgid=monthsIds[month], domain="plonelocales", context=context, default=monthsIds[month], target_language='fr')
+        #use mOntH here to avoid strftime translating the month...
+        res = date.strftime('%e mOntH %Y' + hour)
+        #replace '01 ' by '1er '
+        res = res.replace('01 ', '1er ')
+        #... now that we have mOntH in the string, we can easily replace it
+        res = res.replace('mOntH', translated_month.lower())
+        return res.strip()
+    else:
+        return date.strftime("%d/%m/%Y" + hour)
+
+def wordizeDate(date, context=None, long_format=False):
+    """
+      Convert a date made of digits to a date made of words.
+      If long_format is True, we convert the hour to words too
+    """
+    converted_month = formatDate(date, month_name=True, context=context, long_format=False).split(' ')[1]
+
+    splitted_date = date.strftime("%Y/%m/%d").split('/')
+    date_time = ""
+    #check if we have a time
+    if len(str(date).split(' ')) > 1:
+        date_time = str(date).split(' ')[1]
+
+    #work on the date
+    day_to_convert = int(splitted_date[2])
+    if day_to_convert == 1:
+        converted_day = "premier"
+    else:
+        converted_day = int2word(day_to_convert)
+    converted_year = int2word(splitted_date[0])
+    converted_date = "%s %s %s" % (converted_day, converted_month, converted_year)
+
+    converted_time = ""
+    if long_format and date_time:
+        #work on the hour
+        splitted_time = date_time.split(':')
+        hour = splitted_time[0]
+        minutes = splitted_time[1]
+        converted_hour = int2word(int(hour))
+        if converted_hour.endswith("un "):
+            converted_hour = converted_hour[:-1] + "e"
+        converted_minutes = int2word(int(minutes))
+        if hour in ['01', '1', '0', '00', ]:
+            hour_word = "heure"
+        else:
+            hour_word = "heures"
+        #do not display 'zero' in the date like : eleven hour zero...
+        #see http://dev.communesplone.org/trac/ticket/1553
+        if converted_minutes and not minutes in ['0', '00',]:
+            converted_time = " %s %s %s %s" % (u"à", converted_hour, hour_word, converted_minutes)
+        else:
+            #remove the minutes, especially useless spaces if there are no minutes
+            converted_time = " %s %s %s" % (u"à", converted_hour, hour_word)
+    return converted_date.encode('utf-8') + converted_time.encode('utf-8')
+
+def int2word(n):
+    """
+      Convert an integer number into a string of french word
+      Based on the http://www.daniweb.com/code/snippet609.html
+      code that convert into english words...
+    """
+    ############# globals ################
+    ones = ["", "un ","deux ","trois ","quatre ", "cinq ",
```

Comment:
Please replace constants using uppercased variables (ONES, TENS, ...) and adapt the code